test/testUnregisterMachine.js

```diff
@@ -67,26 +92,29 @@ describe('mercurius unregisterMachine', function() {
       token: token,
       machineId: 'machine_2',
     })
-    .expect(200, done);
+    .expect(404, done);
   });
 
-  it('replies with 404 when trying to unregister a non existing token', function(done) {
-    request(mercurius.app)
-    .post('/unregisterMachine')
-    .send({
-      token: 'nonexistingtoken',
-      machineId: 'machine',
+  it('deletes machine\'s clients object after removing the machine', function(done) {
+    redis.exists('machine_1:clients')
+    .then(function(exists) {
+      exists.should.equal(0);
+      done();
     })
-    .expect(404, done);
+    .catch(done);
   });
 
-  it('replies with 404 when trying to unregister a non registered machine', function(done) {
-    request(mercurius.app)
-    .post('/unregisterMachine')
-    .send({
-      token: token,
-      machineId: 'non-existing-machine',
+  it('deletes token after unregistering the last machine', function(done) {
```

Comment:
We should also test that it doesn't delete the token after unregistering a machine that isn't the last one.